### DISCUSSION TITLE
Support for other architectures with seccomp?

### DISCUSSION THREAD

**Discussion by NattyNarwhal** [lldpd/lldpd#677] [General]
I was building lldpd on a ppc64 system, but ran into an issue with the seccomp support. It seems that it only supports x86 systems, because it inspects architecture-dependent registers. I made a note of this to [downstream Gentoo in their bug tracker](https://bugs.gentoo.org/940014), and they've made it so the package doesn't build with seccomp outside of x86.

However, it would be ideal if the seccomp stuff worked across other architectures. While adding more architectures to the current setup is possible, it doesn't feel ideal as it is. Unfortunately, I don't know what would be much better, considering the kernel documentation itself points to an example that tells you to do it like this (as mentioned in the source).

**Comment by vincentbernat** [comment]
It could be made to work with more architectures as the register is only needed to make the log message more informative. I am more worried about enabling seccomp on totally untested architectures, as the syscalls needed may be different. Therefore, we would need more tests.

**Reply by NattyNarwhal** [reply]
Yeah, the fact as seccomp relies on predicting libc's behaviour on some architectures (and the fact syscalls are different per architecture on Linux) as the downstream issue suggests makes this unfortunately really annoying.
